src/contracts/digital-twin/digital-twin.spec.ts

```diff
@@ -147,6 +148,25 @@ describe('DigitalTwin', function test() {
       favorites = await DigitalTwin.getFavorites(runtime);
       expect(favorites).to.not.include(await twin.getContractAddress());
     });
+
+    it('automatically creates a valid did document upon twin creation', async () => {
+      const twin = await DigitalTwin.create(runtime, defaultConfig);
+      const twinIdentity = (await twin.getDescription()).identity;
+      const did = await runtime.did.convertIdentityToDid(twinIdentity);
+      const ownerIdentity = await runtime.verifications
+        .getIdentityForAccount(defaultConfig.accountId, true);
+      const ownerDid = await runtime.did.convertIdentityToDid(ownerIdentity);
+      const ownerDidDocument = await runtime.did.getDidDocument(ownerDid);
+
+      const promise = runtime.did.getDidDocument(did);
+
+      expect(ownerDidDocument).not.to.be.null;
+      expect(promise).not.to.be.rejected;
+      expect((await promise).id).to.eq(did);
```

Comment:
not sure if it's a good idea to put the await promise in closure here.
you could use `.eventually` keyword for check results of promises, see: https://www.chaijs.com/plugins/chai-as-promised/